eureka2-client/src/main/java/com/netflix/eureka2/client/resolver/RoundRobinServerResolver.java

```diff
@@ -0,0 +1,143 @@
+package com.netflix.eureka2.client.resolver;
+
+import com.netflix.eureka2.Server;
+import com.netflix.eureka2.interests.ChangeNotification;
+import com.netflix.eureka2.interests.ChangeNotifications;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Notification;
+import rx.Observable;
+import rx.functions.Func1;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A Server Resolver that uses a round-robin strategy to select between servers. The {@link #resolve()} will
+ * not emit any elements until the resolver have warmed up for the first time, i.e. the resolver has been loaded
+ * with an initial list of servers. The warm up time can be changed from defaults by calling
+ * {@link #withWarmUpConfiguration(int, TimeUnit)}.
+ *
+ * @author David Liu
+ */
+public class RoundRobinServerResolver implements ServerResolver {
+
+    private static final Logger logger = LoggerFactory.getLogger(RoundRobinServerResolver.class);
+
+    private static final Exception SERVER_CACHE_EMPTY_EXCEPTION = new NoSuchElementException("No servers available for this resolver");
+
+    private final int warmUpTimeout;
+    private final TimeUnit timeUnit;
+
+    private final Observable<ChangeNotification<Server>> serverSource;
+    private final AtomicReference<List<Server>> serverCacheRef;
+    private final AtomicInteger positionRef;
+
+    protected RoundRobinServerResolver(Server... servers) {
+        this(sourceFromList(servers), 10, TimeUnit.SECONDS);
+    }
+
+    protected RoundRobinServerResolver(Observable<ChangeNotification<Server>> serverSource) {
```

Comment:
Warmup time was a workaround needed prior to introducing buffer sentinels. As we have those now, I think we do not need it anymore.